src/SSEConnection.js

```diff
@@ -0,0 +1,82 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+import { List } from "immutable";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of SSEConnection
+   * @constructor
+   * @param {string} url – URL to connect to
+   * @param {object} [options]
+   * @param {boolean} [options.withCredentials=true] – used method
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    Object.defineProperty(this, "_data", {
+      value: List(),
+      writable: true
+    });
+
+    Object.defineProperty(this, "options", {
+      value: { withCredentials: options.withCredentials || false }
+    });
+  }
+
+  /**
+   * Returns the data received so far
+   * @returns List<String>
+   */
+  get data() {
+    return this._data;
+  }
+
+  /**
+   * Opens the connection to the configured URL
+   */
+  open() {
+    const { withCredentials } = this.options;
+    this.source = new window.EventSource(this.url, { withCredentials });
+
+    this.source.onopen = () => {
+      this.state = SSEConnection.OPEN;
+      this.emit(
+        ConnectionEvent.OPEN,
+        new ConnectionEvent(this, ConnectionEvent.OPEN)
+      );
+    };
+
+    this.source.onmessage = response => {
+      this._data = this._data.push(response);
+      this.emit(
```

Comment:
So there's no way of getting the data message by message, only the whole list every time?

Reply:
I would have loved to do a rxjs observable here, but we discussed offline to keep it simple. Would be ok with both